nixos/modules/services/mail/postfix.nix

```diff
@@ -771,18 +767,13 @@ in
         recipient_canonical_classes = [ "envelope_recipient" ];
       }
       // optionalAttrs cfg.enableHeaderChecks { header_checks = [ "regexp:/etc/postfix/header_checks" ]; }
-      // optionalAttrs (cfg.sslCert != "") {
-        smtp_tls_CAfile = cfg.sslCACert;
-        smtp_tls_cert_file = cfg.sslCert;
-        smtp_tls_key_file = cfg.sslKey;
-
-        smtp_use_tls = true;
-
-        smtpd_tls_CAfile = cfg.sslCACert;
-        smtpd_tls_cert_file = cfg.sslCert;
-        smtpd_tls_key_file = cfg.sslKey;
-
-        smtpd_use_tls = true;
```

Comment:
Maybe we should add an option to set `smtpd_tls_security_level`. Seems like a very important option that users of the module should be made aware of.

Reply:
"may" is a sane default, but not the upstream one. So I would be hesitant to make it the default in NixOS.
If @asbachb wants to add a commit adding such an option, I don't see a need for a separate PR. Otherwise, I'll make a PR with the change.

Reply:
@dotlambda Only because TLS is disabled by default. When using it, "may" is the level they tell you to use: http://www.postfix.org/TLS_README.html#server_enable

> Explicitly switch it on with "smtpd_tls_security_level = may".

The only other options are "none", and "encrypt". For "encrypt", it has the following to say:

> According to RFC 2487 this MUST NOT be applied in case of a publicly-referenced SMTP server. Instead, this option should be used only on dedicated servers.

Per RFC 2487, a "publicly-referenced SMTP server" is:

> A publicly-referenced SMTP server MUST NOT require use of the
> STARTTLS extension in order to deliver mail locally. This rule
> prevents the STARTTLS extension from damaging the interoperability of
> the Internet's SMTP infrastructure. A publicly-referenced SMTP server
> is an SMTP server which runs on port 25 of an Internet host listed in
> the MX record (or A record if an MX record is not present) for the
> domain name on the right hand side of an Internet mail address.

In other words, unless you're running Postfix for an intranet, you almost certainly want to use "may" here.